starlite/events/listener.py

```diff
@@ -0,0 +1,38 @@
+from typing import TYPE_CHECKING, Any, List, Union
+
+from starlite.exceptions import ImproperlyConfiguredException
+from starlite.utils import AsyncCallable
+
+if TYPE_CHECKING:
+    from starlite.types import AnyCallable
+
+
+class EventListener:
+    """Decorator for event listeners"""
+
+    __slots__ = ("event_ids", "fn", "listener_id")
+
+    fn: "AsyncCallable[Any, Any]"
+
+    def __init__(self, event_id: Union[str, List[str]]):
```

Comment:
```suggestion
    def __init__(self, event_id: Union[str, Iterable[str]]):
```
`List` is unnecessarily restrictive, all we do is iterate over the values to construct the mapping in the emitter, so it could just as well be an iterator, or sequence of strings if order is important.

Reply:
@provinzkraut's arg packing suggestion is even better.